### DISCUSSION TITLE
Methods for leveraging configured search client/indices.

### DISCUSSION THREAD

**Discussion by timkelty** [studioespresso/craft-scout#198] [General]
I want to call arbitrary `Algolia\AlgoliaSearch\SearchIndex` methods (e.g. via CLI command), but want to leverage Scout's settings for configuring the client/index.

Related: https://github.com/riasvdv/craft-scout/pull/122

**Comment by timkelty** [comment]
I just realized that Scout sets a singleton with the configured `SearchClient`, so I can get the client via `Craft::$container->get` and do what I need like so:

```php
<?php

namespace modules\appmodule\console\controllers;

use Algolia\AlgoliaSearch\SearchClient;
use Craft;
use rias\scout\Scout;
use Tightenco\Collect\Support\Collection;
use yii\console\ExitCode;
use yii\helpers\Console;

class AlgoliaController extends \craft\console\Controller
{
    public function actionRandomizeProducts()
    {
        $indexName = getenv('ALGOLIA_INDEX_PREFIX') . 'products';
        $scoutSettings = Scout::$plugin->getSettings();
        $scoutIndex = $scoutSettings->getIndices()
            ->firstWhere('indexName', $indexName);

        $index = Craft::$container->get(SearchClient::class)->initIndex($indexName);
        $ids = $scoutIndex->criteria->ids();
        $updates = Collection::make($ids)
          ->map(function ($id) use ($ids) {
              return [
                'objectID' => $id,
                'random' => rand(0, count($ids)),
            ];
          });

        $index->partialUpdateObjects($updates->all());

        return ExitCode::OK;
    }
}
```

…seems like alot of duplicated effort though. Seems like I should be able to get the `Algolia\AlgoliaSearch\SearchIndex` via the `ScoutIndex` maybe?

**Comment by riasvdv** [comment]
I'd be happy with a method `getClient` on the engine that resolves it from the container, and then the `__call` method  can use that getClient method instead of `$this->algolia`

**Comment by timkelty** [comment]
thanks @riasvdv, I'll amend this PR.
